pulldown-cmark-mdcat/src/render/data.rs

```diff
@@ -39,6 +39,70 @@ impl CurrentLine {
     }
 }
 
+/// A cell in the table.
+#[derive(Debug)]
+pub struct TableCell<'a> {
+    pub(super) fragments: Vec<CowStr<'a>>,
+}
+
+/// A row in the table.
+#[derive(Debug)]
+pub struct TableRow<'a> {
+    pub(super) cells: Vec<TableCell<'a>>,
+    pub(super) current_cell: TableCell<'a>,
+}
+
+/// The state of the current table.
+#[derive(Debug)]
+pub struct CurrentTable<'a> {
+    /// Rows of the table
+    pub(super) rows: Vec<TableRow<'a>>,
+    pub(super) current_row: TableRow<'a>,
+}
+
+impl<'a> TableCell<'a> {
+    pub(super) fn empty() -> Self {
+        TableCell {
+            fragments: Vec::new(),
+        }
+    }
+}
+
+impl<'a> TableRow<'a> {
```

Comment:
I'd appreciate if you'd place these impls right after the corresponding struct definition.

That's just how I prefer it in rust code 😇